hudi-common/src/main/java/org/apache/hudi/common/table/view/IncrementalTimelineSyncFileSystemView.java

```diff
@@ -159,7 +177,12 @@ private void runIncrementalSync(HoodieTimeline timeline, TimelineDiffResult diff
             } else if (instant.getAction().equals(HoodieTimeline.ROLLBACK_ACTION)) {
               addRollbackInstant(timeline, instant);
             } else if (instant.getAction().equals(HoodieTimeline.REPLACE_COMMIT_ACTION)) {
-              addReplaceInstant(timeline, instant);
+              boolean isClusteringCommit = ClusteringUtils.isClusteringCommit(metaClient, instant);
+              if (!instant.isCompleted() && isClusteringCommit) {
+                addPendingClusteringInstant(instant);
+              } else {
+                addReplaceInstant(timeline, instant);
```

Comment:
in L163, we already filter only for clustering replace commits. So, wouldn't this mean, addCompletedClusteringInstant. 
or am I mis-understanding something ?